pkg/guacrest/server/server.go

```diff
@@ -36,10 +37,34 @@ func (s *DefaultServer) HealthCheck(ctx context.Context, request gen.HealthCheck
 }
 
 func (s *DefaultServer) AnalyzeDependencies(ctx context.Context, request gen.AnalyzeDependenciesRequestObject) (gen.AnalyzeDependenciesResponseObject, error) {
-	//return gen.AnalyzeDependencies200JSONResponse{
-	//  PurlListJSONResponse: []gen.Purl{},
-	//}, nil
-	return nil, fmt.Errorf("Unimplemented")
+	if request.Params.Sort == "frequency" {
+		deps := dependencies.New(ctx, s.gqlClient)
+		packages, err := deps.GetSortedDependents()
+		if err != nil {
+			return gen.AnalyzeDependencies500JSONResponse{
+				InternalServerErrorJSONResponse: gen.InternalServerErrorJSONResponse{
+					Message: fmt.Sprintf("failed to get sorted dependents %v", err.Error()),
+				},
+			}, fmt.Errorf("failed to get sorted dependencies %v", err)
+		}
+
+		var packageNames []gen.PackageName
+
+		for _, p := range packages {
+			pac := p // have to do this because we don't want packageNames to keep on appending a pointer of the same variable p.
+			packageNames = append(packageNames, gen.PackageName{
+				Name:           &pac.Name,
+				DependentCount: &pac.DependentCount,
+			})
+		}
+
+		val := gen.AnalyzeDependencies200JSONResponse{
+			PackageNameListJSONResponse: packageNames,
+		}
+
+		return val, nil
+	}
+	return gen.AnalyzeDependencies200JSONResponse{PackageNameListJSONResponse: []gen.PackageName{}}, nil
```

Comment:
I think it would be good to also check if the argument is one of the supported enums.  For example, if the argument is `sort=foo`, then the output would be "scorecard sort is unimplemented".

Reply:
The code is now using a switch case to check for the scorecard sort, or any other type of sort.